src/Toolbar.tsx

```diff
@@ -1,19 +1,451 @@
-import React, { useState } from 'react';
+/* eslint-disable no-multi-assign */
+/* eslint-disable prefer-destructuring */
+import React, { useRef, useState, useEffect } from 'react';
+import styled from '@emotion/styled';
+import { css } from '@emotion/react';
+import { BiBold, BiUnderline, BiItalic, BiLink } from 'react-icons/bi';
 
 interface ToolbarProps {
   containerRef: React.MutableRefObject<HTMLDivElement | null>;
 }
 
-const Toolbar = ({ containerRef }: ToolbarProps) => {
-  const openTitleMenu = () => {};
+function Toolbar({ containerRef }: ToolbarProps) {
+  const [toolbarPosition, setToolbarPosition] = useState<[number, number]>([0, 0]);
+  const [showToolbar, setShowToolbar] = useState<boolean>(false);
+  const toolbarRef = useRef<HTMLDivElement | null>(null);
 
-  const setBold = () => {};
+  useEffect(() => {
+    document.addEventListener('pointerup', (e) => {
+      const selection: Selection | null = window.getSelection();
 
-  const setUnderLine = () => {};
+      if (selection && selection.type === 'Caret') {
+        setShowToolbar(false);
+      }
 
-  const setItalic = () => {};
+      if (selection && selection.type === 'Range') {
+        const range = selection.getRangeAt(0);
+        if (containerRef.current && containerRef.current.contains(range.commonAncestorContainer)) {
+          calculateToolBarPosition();
+        } else {
+          setShowToolbar(false);
+        }
+      } else {
+        setShowToolbar(false);
+      }
+    });
+  }, [containerRef]);
 
-  return <div>Toolbar</div>;
-};
+  const calculateToolBarPosition = () => {
+    const selection: Selection | null = window.getSelection();
+
+    if (selection && selection.type === 'Range') {
+      const range = selection.getRangeAt(0);
+      const rect = range.getBoundingClientRect();
+      setToolbarPosition([rect.left, window.scrollY + rect.bottom]);
+      setShowToolbar(true);
+    } else {
+      setShowToolbar(false);
+    }
+  };
```

Comment:
```suggestion
  const changeToolBarPosition = () => {
    const selection: Selection | null = window.getSelection();

    if (selection?.type !== 'Range') {
      setShowToolbar(false);
      return;
    }
    
    const range = selection.getRangeAt(0);
    const rect = range.getBoundingClientRect();
    
    setToolbarPosition([rect.left, window.scrollY + rect.bottom]);
    setShowToolbar(true);
  };
```

state를 set하는 함수라면 이름 앞에 change 라고 명명하면 어떨까요?
calculate 라는 이름은 출력하는 값이 필요할 것 같습니다.

Reply:
추가로 gaurd clauses 로 들여쓰기 형태를 줄일 수 있습니다.